api/server/agent.go

```diff
@@ -22,9 +23,12 @@ func (tapi ToDDApi) Agent(w http.ResponseWriter, r *http.Request) {
 	// Retrieve query values
 	values := r.URL.Query()
 
-	var agent_list []defs.AgentAdvert
-
-	var tdb = db.NewToddDB(tapi.cfg)
+	agent_list, err := tapi.tdb.GetAgents()
```

Comment:
Given that you're invoking a 500, that's good enough for me. The client should be able to detect this and display a useful error message. Thanks!